src/Tool/ONNXMLIROpt/ONNXMLIROpt.cpp

```diff
@@ -64,6 +64,30 @@ static llvm::cl::opt<bool> allowUnregisteredDialects(
     llvm::cl::desc("Allow operation with no registered dialects"),
     llvm::cl::init(false));
 
+enum OptLevel { O0 = 0, O1, O2, O3 };
+static llvm::cl::opt<OptLevel> OptimizationLevel(
+    llvm::cl::desc("Optimization levels:"),
+    llvm::cl::values(clEnumVal(O0, "Optimization level 0 (default)."),
+        clEnumVal(O1, "Optimization level 1."),
+        clEnumVal(O2, "Optimization level 2."),
+        clEnumVal(O3, "Optimization level 3.")),
+    llvm::cl::init(O0));
+
+void scanAndSetOptLevel(int argc, char **argv) {
+  // In decreasing order, so we pick the last one if there are many.
+  for (int i = argc - 1; i > 0; --i) {
+    std::string currStr(argv[i]);
+    if (currStr.find("-O") == 0) {
+      int num = atoi(&argv[i][2]); // Get the number starting 2 char down.
+      // Silently ignore out of bound opt levels.
```

Comment:
It would be nice to emit an error message if the optlevel is incorrect (and abort).

Reply:
It will happen as the option is still scanned later by the regular code. I did not want to double check it here. Let me know if you still feel that way.
``` bash
root@1250b049ac2b:~/workspace/atests/ops# onnx-mlir-opt conv2.mlir -shape-inference -convert-onnx-to-krnl -canonicalize -O7
onnx-mlir-opt: Unknown command line argument '-O7'.  Try: 'onnx-mlir-opt --help'
onnx-mlir-opt: Did you mean '--O0'?
```

Reply:
Good point, I am fine with the code as is given a diagnostic message will be emitted.